File: agent_environment/misc.js
const fs = require("fs");

exports.loadConfig = () => {
    if (process.argv.length > 2) {
        const configPath = process.argv[2];
        const text = fs.readFileSync(configPath);
        const config = JSON.parse(text);
        return config;
    } // end if
    return defaultConfig; 
} // end loadConfig

exports.showConfig = (config) => {
    console.log(`Address to connect is ${config.brokerAddress}`)
    if (config.id) {
        console.log(`Agent with id ${config.id}`);
    } else {
       console.log(`Environment with these pairs set up: ${JSON.stringify(config.pairs)}`) 
    } // end else
} // end showConfig

const defaultConfig = {
    "brokerAddress": "mqtt://localhost:1883",
} // end defaultConfig
 

File: agent_environment/misc.py
import sys, subprocess
import json


defaultConfig = {
    "brokerAddress": "mqtt://localhost:1883",
}

def show_config(config):
    address = config["brokerAddress"]
    id = False 
    try:
        id = config["id"]
    except Exception as e:
        pass
    
    print(f"MQTT Broker address is {address}")
    if id:
        print(f"This is an Agent with id {id}")
    else: 
        pairs = config["pairs"]
        print(f"This is an Environment with these pairs set up: {json.dumps(pairs)}")


def load_config():
    args = sys.argv
    if len(args) > 1:
        config_path = args[1]
        text = ""
        with open(config_path, "r") as file:
            text = file.readlines()
            text = "".join(text)
        config = json.loads(text)
        return config

    return defaultConfig



def run_clingo(files: list):
    command = ["clingo"]
    command.extend(files)
    command.extend(["-V0", "--out-atom=%s."]) 
    result= subprocess.run(command, capture_output=True)
    return result
    
